js/src/index.js

```diff
@@ -219,6 +219,48 @@ operators.$map = (template, context) => {
   }
 };
 
+operators.$reduce = (template, context) => {
+  const EACH_RE = 'each\\(([a-zA-Z_][a-zA-Z0-9_]*),\\s*([a-zA-Z_][a-zA-Z0-9_]*)(,\\s*([a-zA-Z_][a-zA-Z0-9_]*))?\\)';
+  checkUndefinedProperties(template, ['\\$reduce', 'initial', EACH_RE]);
+  let value = render(template['$reduce'], context);
+  if (!isArray(value) && !isObject(value)) {
+    throw new TemplateError('$reduce value must evaluate to an array or object');
+  }
+
+  if (Object.keys(template).length !== 3) {
+    throw new TemplateError('$reduce must have exactly three properties');
+  }
+
+  let eachKey = Object.keys(template).filter(k => k !== '$reduce' && k !== 'initial')[0];
```

Comment:
I suppose you could just use a find here

```suggestion
  let eachKey = Object.keys(template).find(k => k !== '$reduce' && k !== 'initial');
```